### ISSUE TITLE
NullpointerException on filtering datalab columns that have empty attribute values

### ISSUE THREAD

**Issue by jbliesener** [gephi/gephi#2015]
## Expected Behavior
Using the "filter" textbox in datalab should work even when there are missing attribute values in some rows of the column data

(retroactively changed wording for clarification) 

## Current Behavior
```
NullPointerException
java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Matcher.java:1283)
	at java.util.regex.Matcher.reset(Matcher.java:309)
	at java.util.regex.Matcher.reset(Matcher.java:329)
	at javax.swing.RowFilter$RegexFilter.include(RowFilter.java:448)
	at javax.swing.RowFilter$GeneralFilter.include(RowFilter.java:414)
	at javax.swing.DefaultRowSorter.include(DefaultRowSorter.java:931)
	at javax.swing.DefaultRowSorter.initializeFilteredMapping(DefaultRowSorter.java:640)
	at javax.swing.DefaultRowSorter.sort(DefaultRowSorter.java:592)
	at javax.swing.DefaultRowSorter.setRowFilter(DefaultRowSorter.java:424)
	at org.jdesktop.swingx.JXTable.setRowFilter(JXTable.java:1910)
	at org.gephi.desktop.datalab.tables.AbstractElementsDataTable.setFilterPattern(AbstractElementsDataTable.java:229)
	at org.gephi.desktop.datalab.DataTableTopComponent.refreshAppliedFilter(DataTableTopComponent.java:486)
	at org.gephi.desktop.datalab.DataTableTopComponent.access$500(DataTableTopComponent.java:134)
...

```
## Possible Solution
Maybe deriving from RegexFilter adding tolerance for null values?

## Steps to Reproduce

1. Create a network manually in datalab, leaving some string attributes empty (do not click in the value field in order to make sure that actually no value is stored)
2. Try filtering in datalab for attributes that don't have values in some of the rows

(retroactively changed wording for clarification)

## Context

I'm creating networks from my own plugin and some values really ARE absent. Whenever that happens, I can't filter the column in datalab anymore.

## Your Environment

* Product Version: Gephi 0.9.2 201709241107
* Java: 1.8.0_161; Java HotSpot(TM) 64-Bit Server VM 25.161-b12
* Runtime: Java(TM) SE Runtime Environment 1.8.0_161-b12
* System: Windows 10 version 10.0 running on amd64; Cp1252; en_US (gephi)

**Comment by jbliesener** [comment]
Clarification: It's not that the attribute value actually is NULL, the attribute is ABSENT. See attached pic and file. Set the value of the Filterdemo attribute of node Id 0 to anything different from null  and it works.

![image](https://user-images.githubusercontent.com/5624993/44230176-cb0f8080-a170-11e8-8c1a-44aa9d1043b3.png)

[Filterdemo.zip](https://github.com/gephi/gephi/files/2294965/Filterdemo.zip)

**Comment by eduramiba** [comment]
Thanks for the report!

**Comment by mihkal** [comment]
one could enable user to add value like 1 for attributes like this?

**Comment by jbliesener** [comment]
Ping. Any chance to fix this in the upcoming release? Shouldn't be a big one...

**Comment by mbastian** [comment]
Hi @jbliesener, we have it labelled for 0.9.3 so we'll look into it

**Comment by jbliesener** [comment]
Great, @mbastian. Thanks a lot for giving us Gephi.

**Comment by mbastian** [comment]
Thanks for the report, this is now fixed. It will be included in the 0.9.3 release.

**Comment by jbliesener** [comment]
Thank you so much, Mathieu.
